### PR TITLE
Release: slate-cbl v2.1.13

### PR DESCRIPTION
**Pull request by nbey**
- Gracefully skip ineligible section enrollees when creating student task assignments.
- Fix issue with Experience Type not displaying in the student task detail window of the `SlateTasksStudent` app.
- Set `re-assigned` tasks that are submitted to `re-submitted`
- Allow guardians to view competency details view in `SlateDemonstrationsStudent` app
- Fix expired tasks nightly cron script
- Exclude non-student and disabled accounts in guardian CBL tool dashboard
- Add `Rated` column to exports

### CODE REVIEW

**Comment by nbey** [comment]
@themightychris Could you review this release?

File: php-classes/Slate/CBL/Demonstrations/DemonstrationSkillsRequestHandler.php [hunk 1]
```diff
@@ -29,8 +29,15 @@ public static function handleBrowseRequest($options = [], $conditions = [], $res
                 return static::throwNotFoundError('Student not found');
             }
 
-            if (!$GLOBALS['Session']->hasAccountLevel('Staff') && $Student->ID != $GLOBALS['Session']->PersonID) {
-                return static::throwUnauthorizedError('Only staff may browse others\' records');
+            if (
+                    !$GLOBALS['Session']->hasAccountLevel('Staff') &&
+                    $Student->ID != $GLOBALS['Session']->PersonID &&
+                    !$GuardianRelationship = \Emergence\People\GuardianRelationship::getByWhere([
```
Comment (themightychris) [inline php-classes/Slate/CBL/Demonstrations/DemonstrationSkillsRequestHandler.php]: @nbey `use` this class at the top of the file to shorten this line and make clear at the top all the external dependencies

Comment (themightychris) [reply php-classes/Slate/CBL/Demonstrations/DemonstrationSkillsRequestHandler.php]: @nbey you need to use `getByClass` or add a `Class` condition to this to ensure that only relationships of type `GuardianRelationship` match, `getByWhere` on its own searches the whole table and doesn't filter by class hierarchy

File: php-classes/Slate/CBL/Demonstrations/DemonstrationSkillsRequestHandler.php [hunk 2]
```diff
@@ -29,8 +29,15 @@ public static function handleBrowseRequest($options = [], $conditions = [], $res
...
+                        'PersonID' => $GLOBALS['Session']->PersonID,
+                        'RelatedPersonID' => $Student->ID
+                    ]))
+                {
```
Comment (themightychris) [inline php-classes/Slate/CBL/Demonstrations/DemonstrationSkillsRequestHandler.php]: @nbey drop the closing paren for the if to its own line, align it with the opening `if`, and then put the opening curly bracket after it like you see here: http://stackoverflow.com/questions/23758916/psr-2-standard-for-long-if-conditions

File: php-classes/Slate/CBL/Tasks/StudentTasksRequestHandler.php [hunk 3]
```diff
@@ -22,6 +22,8 @@ public static function handleRecordsRequest($action = false)
         switch ($action = $action ?: static::shiftPath()) {
             case 'assigned':
                 return static::handleAssignedRequest();
+
+            // todo: re-implement endpoint RE: https://jarvus.atlassian.net/browse/CBL-215
```
Comment (themightychris) [inline php-classes/Slate/CBL/Tasks/StudentTasksRequestHandler.php]: @nbey convention is to capitalize `// TODO:`, leaving it lowercase could lead to this comment not coming up in searches

File: php-classes/Slate/CBL/Tasks/StudentTasksRequestHandler.php [hunk 4]
```diff
@@ -139,11 +141,10 @@ public static function handleStudentTaskSubmissionRequest()
 
         if ($_REQUEST) {
             $student_task = static::getRecordByHandle($_REQUEST['ID']);
-
         }
 
         static::setStudentTaskAttachments($student_task, $_REQUEST);
-        if (!empty($student_task->Submissions)) {
+        if (!empty($student_task->Submissions) || $student_task->TaskStatus == 're-assigned') {
```
Comment (themightychris) [inline php-classes/Slate/CBL/Tasks/StudentTasksRequestHandler.php]: @nbey PHP variables should be cameCase not snake_case

File: php-classes/Slate/CBL/Tasks/TasksRequestHandler.php [hunk 5]
```diff
@@ -265,8 +264,12 @@ protected static function onRecordSaved(\ActiveRecord $Record, $data)
...
                     'ExpirationDate' => $Record->ExpirationDate
                 ]);
-
-                $studentTask->save(false);
+                try {
+                    $studentTask->save(false);
+                } catch (\RecordValidationException $e) {
+                    // record failed validation, continue creating others.
+                    continue;
+                }
```
Comment (themightychris) [inline php-classes/Slate/CBL/Tasks/TasksRequestHandler.php:273]: @nbey put some blank lines around blocks like this

File: php-config/Slate/UI/Tools.config.d/cbl.php [hunk 6]
```diff
@@ -29,8 +29,10 @@
 
 if ($GLOBALS['Session']->Person && !empty($GLOBALS['Session']->Person->Wards)) {
     foreach($GLOBALS['Session']->Person->Wards as $Ward) {
-        $cblTools[$Ward->FirstNamePossessive . ' Competency Dashboard'] = '/cbl/dashboards/demonstrations/student?student='.$Ward->Username;
-        $cblTools[$Ward->FirstNamePossessive . ' Task Dashboard'] = '/cbl/dashboards/tasks/student?student='.$Ward->Username;
+        if ($Ward instanceof \Slate\People\Student && $Ward->AccountLevel != 'Disabled') {
```
Comment (themightychris) [inline php-config/Slate/UI/Tools.config.d/cbl.php]: @nbey looks good but it's better style to have one or more "guard clauses" at the top of a block than to deeply nest the whole body https://sourcemaking.com/refactoring/replace-nested-conditional-with-guard-clauses

When you find yourself wrapping the whole body of a function/loop in conditions, invert the conditions and use them to return/continue at the top instead

**Review by themightychris — CHANGES_REQUESTED** [review]

**Pull request merged**